### PR TITLE
docs: Add docs page for experimental flags

### PR DESCRIPTION
**Pull request by sidharthachatterjee**
We should track these because it's easy to forget how many we have 😅 

This list is _not_ complete at the moment but is more of a starting point. Let's iterate and add all of the ones we know as we move forward.

### CODE REVIEW

**Comment by pieh** [comment]
Should this be in sidebar? 

Should we mention that experimental features can have bugs and can change behaviours at any time, as well as be removed without the need for major version bump, etc (but we will try not to do it)

**Comment by sidharthachatterjee** [comment]
> Should this be in sidebar?

The main goal of this at the moment is tracking them for our use. So I suppose this could happen later _if_ we want to explicitly make the page more visible. 

> Should we mention that experimental features can have bugs and can change behaviours at any time, as well as be removed without the need for major version bump, etc (but we will try not to do it)

Not necessary for now since this page is not going to be explicitly in the side bar or any TOC

**Comment by sidharthachatterjee** [comment]
> Not necessary for now since this page is not going to be explicitly in the side bar or any TOC

Went ahead and added it anyway

**Comment by wardpeet** [comment]
Do we really need/want this? Unsure if it gives value to us, at this moment. Why not put this in gatsby's readme?
If so we also have `GATSBY_GRAPHQL_IDE=playground`

**Comment by sidharthachatterjee** [comment]
> Do we really need/want this? Unsure if it gives value to us, at this moment. Why not put this in gatsby's readme?

Good question. I think there's value in listing them in one place for own our sanity and for new folks who come into the team in the future and might not have context. 

> If so we also have GATSBY_GRAPHQL_IDE=playground

This list is not complete at the moment but is more of a starting point. Let's iterate and add all of the ones we know as we move forward. 😛

**Comment by wardpeet** [comment]
> This list is not complete at the moment but is more of a starting point. Let's iterate and add all of the ones we know as we move forward. 😛

Famous last words, this won't happen for existing ones :)

**Comment by sidharthachatterjee** [comment]
> Famous last words, this won't happen for existing ones :)

You're probably right 😛

**Comment by laurieontech** [comment]
Hmmm, interesting. It's a nice idea, but I worry a bit about maintainability.

Could this work as a blog post? Since that comes with an assumption that it's a snapshot in time?

**Comment by laurieontech** [comment]
@sidharthachatterjee following up on this and the question I posed.

**Comment by marcysutton** [comment]
Following up on this again: I do feel it could be useful to advanced users to mention how to work with experimental flags (such as the Fast Refresh one that was discussed over the weekend). It could be included in the sidebar under [Gatsby Internals](https://www.gatsbyjs.org/docs/gatsby-internals/), or [Gatsby API](https://www.gatsbyjs.org/docs/api-reference/).

To make sure this is maintainable, is there a CLI command or something to list all of the flags available at a given time rather than listing them in the doc itself? 

cc @sidharthachatterjee

**Comment by marcysutton** [comment]
Oh and one more thing–docs need to follow the Style Guide for pronouns, so "we do X" should be changed to "Gatsby does X". https://www.gatsbyjs.org/contributing/gatsby-style-guide/

File: docs/docs/experimental-flags.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+We're always making improvements to Gatsby and sometimes, these improvements might be available behind a feature flag. These flags are typically an environment variable that can be set to turn the feature on when starting up either `gatsby develop` or `gatsby build`.
+
+This page aims to list currently known and supported experimental feature flags in Gatsby v2.
+
+Please note that everything on this page is _experimental_ in that they might or might _not_ make it into a stable version of Gatsby in the future. APIs marked as experimental may also include breaking changes in a patch or minor version bump or be removed and users should only proceed with caution.
+
+## Currently Supported Experimental Flags
+
+### GATSBY_DB_NODES
```
Comment (muescha) [inline docs/docs/experimental-flags.md:13]:
```suggestion
### aGATSBY_DB_NODESa
```

File: docs/docs/experimental-flags.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+In preparation for future versions of Gatsby, we’ve enabled experimental support for a different mechanism for the persistence of nodes: [Loki](https://www.npmjs.com/package/lokijs). You can try this by setting `GATSBY_DB_NODES` to `loki`
```
Comment (muescha) [inline docs/docs/experimental-flags.md:15]: Loki support is dropped with:

- #23838 `chore(gatsby): Drop (experimental) support for Loki`

File: docs/docs/experimental-flags.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+Read up more about this in the [Scaling Issues documentation](https://www.gatsbyjs.org/docs/scaling-issues/#gatsby_db_nodes).
```
Comment (muescha) [inline docs/docs/experimental-flags.md:17]: this section is also dropped with #23838 `chore(gatsby): Drop (experimental) support for Loki`

**Comment by LekoArts** [comment]
@sidharthachatterjee I'll close this one as stale, this should probably be revisited later.

**Pull request closed without merging**